src/object.rs

```diff
@@ -50,6 +50,14 @@ pub trait INSObject : Any + Sized + Message {
             Id::from_retained_ptr(obj)
         }
     }
+    
+    fn alloc() -> Id<Self> {
+        let cls = Self::class();
+        unsafe {
+            let obj: *mut Self = msg_send![cls, alloc];
+            Id::from_retained_ptr(obj)
```

Comment:
This method shouldn't be safe, it returns a partially constructed object and violates the guarantee that an object's designated initializer has been invoked to properly set up its state.

With this method, using only safe code I could write:
``` rust
let string = NSString::alloc();
let len = string.len();
```

And the behavior will be undefined! (Or at least, it could do something pretty unexpected in a lot of cases.)

Even if we mark this method as unsafe, it can still be pretty easy to mess up. Since it returns an `Id<Self>`, it'll automatically be dealloc'd later even if you never initialized it, so maybe it'd be better to return `*mut Self` instead.

At that point, is there much advantage to having this method rather than just doing `msg_send![Self::class(), alloc]`?

Does my reasoning make sense? I agree that copy-pasting the alloc code around isn't fun, but returning an un-initialized object is something I've never done in Objective-C so I wouldn't think it necessary in Rust either. Has it been a big inconvenience in your AppKit wrapper?